src/algebra/star/module.lean

```diff
@@ -33,3 +33,35 @@ def star_linear_equiv (R : Type*) {A : Type*}
 { to_fun := star,
   map_smul' := star_smul,
   .. star_add_equiv }
+
+variables (R : Type*) {A : Type*}
+  [semiring R] [invertible (2 : R)] [star_monoid R] [has_trivial_star R]
+  [add_comm_group A] [module R A] [star_add_monoid A] [star_module R A]
+
+/-- The self-adjoint part of an element of a star module, as a linear map. -/
+@[simps] def self_adjoint_part : A →ₗ[R] self_adjoint A :=
+{ to_fun := λ x, ⟨(⅟2 : R) • (x + star x),
+  by simp only [self_adjoint.mem_iff, star_smul, add_comm,
+                  star_add_monoid.star_add, star_inv', star_bit0,
+                  star_one, star_star, star_inv_of (2 : R), star_trivial]⟩,
+  map_add' := λ x y, by { ext, simp [add_add_add_comm] },
+  map_smul' := λ r x, by { ext, by simp [←mul_smul,
+          show ⅟ 2 * r = r * ⅟ 2, from commute.inv_of_left (commute.one_left r).bit0_left] } }
+
+/-- The skew-adjoint part of an element of a star module, as a linear map. -/
+@[simps] def skew_adjoint_part : A →ₗ[R] skew_adjoint A :=
+{ to_fun := λ x, ⟨(⅟2 : R) • (x - star x),
+    by simp only [skew_adjoint.mem_iff, star_smul, star_sub, star_star, star_trivial, ←smul_neg,
+                  neg_sub]⟩,
+  map_add' := λ x y, by { ext, simp only [sub_add, ←smul_add, sub_sub_assoc_swap, star_add,
+                                          add_subgroup.coe_mk, add_subgroup.coe_add] },
+  map_smul' := λ r x, by { ext, simp [←mul_smul, ←smul_sub,
+            show r * ⅟ 2 = ⅟ 2 * r, from commute.inv_of_right (commute.one_right r).bit0_right] } }
+
+lemma star_module.eq_self_adjoint_part_add_skew_adjoint_part (x : A) :
+  x = self_adjoint_part R x + skew_adjoint_part R x :=
+begin
+  simp only [smul_sub, self_adjoint_part_apply_coe, smul_add, skew_adjoint_part_apply_coe,
+    add_add_sub_cancel],
+  rw [←smul_add, ←two_smul R, ←mul_smul, inv_of_mul_self, one_smul],
```

Comment:
The goal here is `x = ⅟ 2 • x + ⅟ 2 • x` which I think deserves to be a lemma in its own right.

Reply:
(as a simp-lemma in the other direction)